library/src/main/java/com/getstream/sdk/chat/viewmodel/CreateChannelViewModel.kt

```diff
@@ -0,0 +1,68 @@
+package com.getstream.sdk.chat.viewmodel
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MediatorLiveData
+import androidx.lifecycle.ViewModel
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.Member
+import io.getstream.chat.android.client.models.name
+import io.getstream.chat.android.livedata.ChatDomain
+
+class CreateChannelViewModel(
+        private val chatDomain: ChatDomain = ChatDomain.instance(),
+        private val chatClient: ChatClient = ChatClient.instance()
+) : ViewModel() {
+    private val stateMerger = MediatorLiveData<State>()
+    val state: LiveData<State> = stateMerger
+
+    fun onEvent(event: Event) {
+        if (event is Event.ChannelNameSubmitted) {
+            val channelNameCandidate = event.channelName
+            val isValidName = validateChannelName(channelNameCandidate)
+            if (isValidName) {
+                queryChannel(channelNameCandidate)
+            } else {
+                stateMerger.postValue(State.ValidationError)
+            }
+        }
+    }
+
+    private fun queryChannel(channelName: String) {
+        val channelId: String = channelName.replace(" ".toRegex(), "-").toLowerCase()
+        val members = chatClient.getCurrentUser()?.run {
+            listOf(Member(this))
+        } ?: listOf()
+        val channel = Channel().apply {
+            this.cid = "messaging:$channelId"
+            this.id = channelId
+            this.type = "messaging"
+            this.name = channelName
+            this.members = members
+        }
+        chatDomain.useCases.createChannel.invoke(channel).execute().run {
+            when  {
+                isSuccess -> {
+                    stateMerger.postValue(State.ChannelCreated)
+                }
+                isError -> {
+                    stateMerger.postValue(State.BackendError)
+                }
+            }
+        }
+    }
+
+    private fun validateChannelName(channelNameCandidate: String): Boolean {
+        return channelNameCandidate.isNotEmpty()
+    }
+
+    sealed class State {
+        object ChannelCreated : State()
```

Comment:
Is there any progress state? Or channel is created locally instantly by domain?

Reply:
if device is online there's a call to the backend, as far as I can see. I'll add a loading state.